### PR TITLE
Updates integrator logic per #171. 

### PR DESCRIPTION
**Pull request by CompRhys**
## Summary

The tests of the function do not use the row vector cell so this suggests this is a bug that it's used in the integrators.

see that tests of function don't use the swap axis: https://github.com/Radical-AI/torch-sim/blob/d2cee6628ff3b4a8da7557772cc12a669a0c09b4/tests/test_transforms.py#L267

## Checklist
* [x] Doc strings have been added in the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google).
  Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [x] Tests have been added for any new functionality or bug fixes.
* [x] All linting and tests pass.

### CODE REVIEW

**Comment by CompRhys** [comment]
~~True error still not showing up with lazy import route, pyi needed to get the language server to work with the lazy imports.~~ reset and moved lazy imports to #179

**Comment by CompRhys** [comment]
#17 is also related to this as if we solve that we can just drop the call to roll the positions back into the cell from the integrators.

**Comment by janosh** [comment]
my preference is to tackle that then and address #17 here instead of merging just a minor patch

**Comment by CompRhys** [comment]
I would still merge this but just wanted to raise that to bump the idea

**Comment by janosh** [comment]
i think i found 2 more bugs in `transforms.py` were the code was still written to work with column vector lattices while the rest of torch-sim now uses row vector convention as of https://github.com/Radical-AI/torch-sim/pull/112


need some eyes on the changes in [6b82750](https://github.com/Radical-AI/torch-sim/pull/175/commits/6b82750f050f2a7b94f738c02078fb3c7b221621) [1e7290e](https://github.com/Radical-AI/torch-sim/pull/175/commits/1e7290e1e472fb0a7499f6c731621460b9f53553) from preferably two reviewers @orionarcher @abhijeetgangan @CompRhys

File: tests/test_transforms.py [hunk 1]
```diff
@@ -99,25 +99,30 @@ def test_pbc_wrap_general_triclinic() -> None:
...
     )
 
     # Position outside triclinic box
-    positions = torch.tensor([[2.5, 2.5, 2.5]])
+    positions = torch.tensor([[2.5, 2.5, 2.5]], dtype=torch.float64)
 
-    # Correct expected wrapped position for this triclinic cell
-    expected = torch.tensor([[2.0, 0.5, 0.2]])
+    # Expected wrapped position calculated using r' = (r @ inv(M_row.T) % 1) @ M_row
+    expected = torch.tensor([[1.875, 0.9875, 0.125]], dtype=torch.float64)
```
Comment (CompRhys) [inline tests/test_transforms.py]: These weird values for the expected result seems self-evidently to suggest broken logic to me? These values must be projected onto the lattice vectors which isn't what I would expect the function to do

Comment (CompRhys) [reply tests/test_transforms.py]:
```py
import numpy as np
from pymatgen.core import Lattice, Structure

lattice = Lattice(np.array([[2.0, 0.5, 0.0],
                   [0.0, 2.0, 0.0],
                   [0.0, 0.3, 2.0]]).T)

# Create structure with a single H atom at the given position
s = Structure(lattice, ["H"], [[2.5, 2.5, 2.5]], coords_are_cartesian=True)

# Get the wrapped (folded) position in cartesian coordinates
wrapped_pos = s.frac_coords[0] % 1  # fractional coords wrapped to [0,1)
wrapped_cart = lattice.get_cartesian_coords(wrapped_pos)

print(wrapped_cart)
```

Comment (CompRhys) [reply tests/test_transforms.py]: this gives the same answer as the old test. I think that also suggests that there was an issue with the old function as I am giving the lattice as column vectors (due to transpose) to a function that expects row vectors.

**Comment by CompRhys** [comment]
@abhijeetgangan this okay to merge?

**Review by janosh — APPROVED** [review]
looks great! thanks @CompRhys 👍
